src/app/shared/services/iam.service.ts

```diff
@@ -69,6 +70,10 @@ export class IamService {
     return this._iam;
   }
 
+  getAssetById(id) {
+    return this.observableWrapper(this.iam.getAssetById({id}), {message: 'Getting selected asset data...'});
```

Comment:
Is the reason for the `observableWrapper` to be able to put the `loadingService` `show()` and `hide()` around the `source`? Maybe a name like `wrapWithLoadingService` would be better?